src/printer.js

```diff
@@ -3272,6 +3255,19 @@ function printFunctionParams(path, print, options, expandArg, printTypeParams) {
   ]);
 }
 
+function shouldPrintParamsWithoutParens(path, options) {
+  if (options.arrowFunctionParentheses === "always") {
+    return false;
+  }
+
+  if (options.arrowFunctionParentheses === "avoid") {
+    const node = path.getValue();
+    return canPrintParamsWithoutParens(node);
+  }
+
+  return false;
```

Comment:
Probably worth adding a comment indicating that this should be unreachable

Reply:
or just

```js
function shouldPrintParamsWithoutParens(path, options) {
  if (options.arrowFunctionParentheses === "avoid") {
    const node = path.getValue();
    return canPrintParamsWithoutParens(node);
  }
  // possible comment here to explain it :)
  return false;
}
```

Reply:
That works too, though I like the explicitness (and having something to grep for, and having a safe default-fallback).